core/src/com/projectkorra/projectkorra/util/DamageHandler.java

```diff
@@ -120,13 +116,12 @@ private static void ignorePercentageArmorDamage(LivingEntity lent, double percen
 	 * @param event The event we want to modify
 	 */
 	public static void entityDamageCallback(EntityDamageEvent event) {
-		if (!ARMOR_PERCENTAGE_BY_ENTITY_ID.containsKey(event.getEntity().getEntityId())) return;
-		double ignorePercentage = ARMOR_PERCENTAGE_BY_ENTITY_ID.get(event.getEntity().getEntityId());
-		ARMOR_PERCENTAGE_BY_ENTITY_ID.remove(event.getEntity().getEntityId()); // We get rid of the entry, so it doesn't call back future non-ability related damage.
-
-		if (ignorePercentage == 0) return;
-
-		if (event.getEntity() instanceof ArmorStand) return; //ArmorStands produce errors when we modify the armor damage, so ignore them.
+		Entity entity = event.getEntity();
+		int entityId = entity.getEntityId();
+		Double ignorePercentage = ARMOR_PERCENTAGE_BY_ENTITY_ID.remove(entityId); // We get rid of the entry, so it doesn't call back future non-ability related damage.
```

Comment:
Why declare as its `Double` wrapper instead of letting Java auto-unbox?

Reply:
I no longer check if it contains key by default, previously it did contains key, get, and remove, I can replace this with just .remove and allow it to be null if an entity isn't in the map (which is why its boxed). Then I just check against armorstand, wasnt in map (is null), and is 0.

It's an optimization from (key, get, remove) -> (remove), since key and remove both have similar logic/duplicated logic to get